internal/pkg/task/task.go

```diff
@@ -0,0 +1,112 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package task provides methods to run task in different configurations.
+package task
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/copilot-cli/internal/pkg/aws/ec2"
+	"github.com/aws/copilot-cli/internal/pkg/aws/ecs"
+	"github.com/aws/copilot-cli/internal/pkg/aws/resourcegroups"
+	"github.com/aws/copilot-cli/internal/pkg/aws/session"
+)
+
+type vpcGetter interface {
+	GetSubnetIDs(filters ...ec2.Filter) ([]string, error)
+	GetSecurityGroups(filters ...ec2.Filter) ([]string, error)
+}
+
+type resourceGetter interface {
+	GetResourcesByTags(resourceType string, tags map[string]string) ([]*resourcegroups.Resource, error)
+}
+
+type defaultClusterGetter interface {
+	DefaultCluster() (string, error)
+}
+
+type taskRunner interface {
+	RunTask(input ecs.RunTaskInput) ([]string, error)
+}
+
+const (
+	startedBy           = "copilot-task"
+)
+
+var (
+	errNoSubnetFound     = errors.New("no subnets found")
+	fmtTaskFamilyName    = "copilot-%s"
+)
+
+// Task gets information required to run a task.
+type Task struct {
+	count     int
+	groupName string
+
+	vpcGetter      vpcGetter
+	resourceGetter resourceGetter
+	clusterGetter  defaultClusterGetter
+	starter        taskRunner
```

Comment:
See a description on why below:
```suggestion
	Count     int
	GroupName string
	Env    string

	VPCClient      VPCGetter
	ResourcesClient ResourceGetter
	ClusterClient  DefaultClusterGetter
	TaskClient        TaskRunner
```